### ISSUE TITLE
chemical_system generated crystals do not make sense

### ISSUE THREAD

**Issue by Mofahdi** [microsoft/mattergen#96]
I fine-tuned the base model with properties: 1- ltc_avg_log and 2- chemical_system with 200 epochs. However, when I tried to generate structures with the following line:
mattergen-generate $RESULTS_PATH_2 --model_path=$MODEL_PATH --batch_size=32 --checkpoint_epoch=best --properties_to_condition_on="{'ltc_avg_log': 2, 'chemical_system':'B-C-N'}" --diffusion_guidance_factor=2.0 

I got structures that contained other elements that are not B, C, or N. How can I only generate structures that contain B, C, and N elements with varying stoichiometries. 

Here is what a partial output looks like:
    "adapter": {
        "pretrained_name": "mattergen_base",
        "model_path": null,
        "load_epoch": "last",
        "full_finetuning": true,
        "adapter": {
            "_target_": "mattergen.adapter.GemNetTAdapter",
            "property_embeddings_adapt": {
                "ltc_avg_log": {
                    "_target_": "mattergen.property_embeddings.PropertyEmbedding",
                    "name": "ltc_avg_log",
                    "unconditional_embedding_module": {
                        "_target_": "mattergen.property_embeddings.EmbeddingVector",
                        "hidden_dim": 512
                    },
                    "conditional_embedding_module": {
                        "_target_": "mattergen.diffusion.model_utils.NoiseLevelEncoding",
                        "d_model": 512
                    },
                    "scaler": {
                        "_target_": "mattergen.common.utils.data_utils.StandardScalerTorch"
                    }
                },
                "chemical_system": {
                    "_target_": "mattergen.property_embeddings.PropertyEmbedding",
                    "name": "chemical_system",
                    "unconditional_embedding_module": {
                        "_target_": "mattergen.property_embeddings.EmbeddingVector",
                        "hidden_dim": 512
                    },
                    "conditional_embedding_module": {
                        "_target_": "mattergen.property_embeddings.ChemicalSystemMultiHotEmbedding",
                        "hidden_dim": 512
                    },
                    "scaler": {
                        "_target_": "torch.nn.Identity"
                    }
                }
            },
            "atom_type_diffusion": "mask",
            "denoise_atom_types": true,
            "gemnet": {
                "_target_": "mattergen.common.gemnet.gemnet_ctrl.GemNetTCtrl",
                "atom_embedding": {
                    "_target_": "mattergen.common.gemnet.layers.embedding_block.AtomEmbedding",
                    "emb_size": 512,
                    "with_mask_type": true
                },
                "cutoff": 7.0,
                "emb_size_atom": 512,
                "emb_size_edge": 512,
                "latent_dim": 512,
                "max_cell_images_per_dim": 5,
                "max_neighbors": 50,
                "num_blocks": 4,
                "num_targets": 1,
                "otf_graph": true,
                "regress_stress": true,
                "scale_file": "/scratch/amlt_code/mattergen/common/gemnet/gemnet-dT.json",
                "condition_on_adapt": [
                    "ltc_avg_log",
                    "chemical_system"
                ]
            },
            "hidden_dim": 512,
            "property_embeddings": {}

**Comment by Mofahdi** [comment]
Hi @danielzuegner , 

Thanks for fixing it. Just wondering though, should I redownload the mattergen repo to rerun mattergen with the new elements masking feature? because you changed the code

Thanks,
Mohammed

**Comment by danielzuegner** [comment]
you will have to run `git pull` to download the latest changes.
